File: libraries/hlp/hchkl.f
      SUBROUTINE hlp_HCHKL (RECORD, LEVEL, NAME)
*+
*  - - - - - -
*   H C H K L
*  - - - - - -
*
*  Check if the line held in RECORD is the start of a new help level
*  and if so get the level number and the keyword.
*
*  Given (in COMMON):
*     LEVOFF   i         logical level for the current help library
*
*  Given (argument):
*     RECORD   c*(*)     the string to be examined
*
*  Returned:
*     LEVEL    i         level number (-1 if not a keyword record)
*     NAME     c*(*)     level name (only if LEVEL.GE.0)
*
*  The line in RECORD is provisionally assumed to be the start of a new
*  help level if the first space-separated field begins at the start of
*  the line and consists wholly of decimal digits.
*
*  Called:  hlp_SPLIT, hlp_DEC
*
*  P.T.Wallace   Starlink   13 June 1995
*-

      IMPLICIT NONE

      CHARACTER*(*) RECORD
      INTEGER LEVEL
      CHARACTER*(*) NAME

      INCLUDE 'helpic'

      INTEGER IPS,IPF,L,I



*  Locate the first field in RECORD.
      CALL hlp_SPLIT(RECORD,1,IPS,IPF)

*  Does it start at the beginning of the line?
      IF (IPS.EQ.1) THEN

*     Yes: attempt to decode a decimal integer.
         CALL hlp_DEC(RECORD,IPS,L)

*     Was it entirely decimal digits?
         I=IPF+1
         IF (IPS.EQ.I) THEN

*        Yes: return the logical level number.
            LEVEL=L+LEVOFF

*        Obtain the level name if any.
            CALL hlp_SPLIT(RECORD,I,IPS,IPF)
            IF (IPS.GT.0) NAME=RECORD(IPS:IPF)
         END IF
      END IF

      END

File: libraries/hlp/hchkl.c
#include <string.h>
#include "hlpsys.h"
void hlpHchkl ( char *record, int *level, char *name )
/*
**  - - - - - - - - -
**   h l p H c h k l
**  - - - - - - - - -
**
**  Check if the line held in record is the start of a new help level
**  and if so get the level number and the keyword.
**
**  Given (in global data):
**     levoff    int      logical level for the current HELP library
**
**  Given (argument):
**     *record   char     the string to be examined
**
**  Returned:
**     *level    int      level number (-1 if not a keyword record)
**     *name     char     level name (only if level >= 0)
**
**  The line in record is provisionally assumed to be the start of a new
**  HELP level if the first space-separated field begins at the start of
**  the line and consists wholly of decimal digits.
**
**  Called:  hlpSplit, hlpDec
**
**  Last revision:   27 August 1995
**
**  Copyright 1996 P.T.Wallace.  All rights reserved.
*/
{
   int ips, ipf, l, i;

/* Preset level */
   *level = -1;

/* Locate the first field in record */
   hlpSplit ( record, 0, &ips, &ipf );

/* Does it start at the beginning of the line? */
   if ( ips == 0) {

   /* Yes: attempt to decode a decimal integer. */
      l = (int) hlpDec ( record, &ips );

   /* Was it entirely decimal digits? */
      i = ipf + 1;
      if ( ips == i ) {

      /* Yes: return the logical level number. */
         *level = l + levoff;

      /* Obtain the level name if any. */
         hlpSplit ( record, i, &ips, &ipf );
         if ( ips > 0 ) {
            for ( i = 0; i <= ipf - ips; i++ ) name [ i ] = record [ i + ips ];
            name [i] = '\0';
         }
      }
   }
}
